version.py

```diff
@@ -117,3 +120,12 @@ def find_latest_version_with_matching_major(reference_version: semver.VersionInf
             if not latest_candidate or latest_candidate < candidate:
                 latest_candidate = candidate
     return latest_candidate
+
+
+def partition_by_major_and_minor(
+    versions: Iterable[semver.VersionInfo],
+) -> Iterable[Set[semver.VersionInfo]]:
+    partitions = defaultdict(set)
+    for version_info in versions:
+        partitions[f'{version_info.major}.{version_info.minor}'].add(version_info)
+    return partitions.values()
```

Comment:
hmm.. have you considered returning the dict? OTOH, this way, we do not have to put too much thought into the chosen key, as it is not exposed to callers.